### PR TITLE
Fix type of `email_sent` field in `invite_new` command

### LINKED ISSUE #3234: InvalidMessageError: {'email_sent': ['Field may not be null.']}
Sentry Issue: [PARSEC-R1C](https://sentry.io/organizations/scille/issues/3658715312/?referrer=github_integration)

Parsec 2.12.0
Backend 2.13.0rc1

This happens when clicking on `Invite a new device`
```
ValidationError: {'email_sent': ['Field may not be null.']}
  File "parsec\serde\serializer.py", line 34, in load
  File "parsec\serde\schema.py", line 183, in load

InvalidMessageError: {'email_sent': ['Field may not be null.']}
  File "parsec\core\backend_connection\cmds.py", line 127, in _send_cmd
  File "parsec\api\protocol\base.py", line 265, in wrapper
  File "parsec\serde\serializer.py", line 86, in loads
  File "parsec\serde\serializer.py", line 37, in load

Invalid response data
```

Error changes when clicking on `Invite a new device` a second time
```
TransportClosedByPeer: Peer has closed connection
  File "parsec\core\backend_connection\cmds.py", line 120, in _send_cmd
  File "parsec\api\transport.py", line 198, in recv

BackendNotAvailable: Peer has closed connection
  File "parsec\core\backend_connection\expose_cmds.py", line 50, in wrapper
  File "parsec\core\backend_connection\cmds.py", line 432, in invite_new
  File "parsec\core\backend_connection\cmds.py", line 124, in _send_cmd

ValidationError: {'email_sent': ['Field may not be null.']}
  File "parsec\serde\serializer.py", line 34, in load
  File "parsec\serde\schema.py", line 183, in load

InvalidMessageError: {'email_sent': ['Field may not be null.']}
  File "parsec\core\backend_connection\cmds.py", line 127, in _send_cmd
  File "parsec\api\protocol\base.py", line 265, in wrapper
  File "parsec\serde\serializer.py", line 86, in loads
  File "parsec\serde\serializer.py", line 37, in load

Invalid response data
```

### PR DESCRIPTION
**Pull request by vxgmichel**
Fix #3234

### CODE REVIEW

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: parsec/backend/invite.py [hunk 1]
```diff
@@ -434,33 +403,67 @@ def _to_http_redirection_url(
...
+                to_addr=to_addr,
+                message=message,
+            )
+        except InvitationEmailRecipientError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.BAD_RECIPIENT)
+        except InvitationEmailConfigError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
+        except Exception as exc:
+            # Fail-safe: since the device/user has been created, we don't want to fail too hard
+            logger.warning("Unexpected exception while sending an email", exc_info=exc)
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/backend/invite.py]: done.

**Review by touilleMan — APPROVED** [review]

File: parsec/backend/invite.py [hunk 2]
```diff
@@ -434,33 +403,67 @@ def _to_http_redirection_url(
...
+                message=message,
+            )
+        except InvitationEmailRecipientError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.BAD_RECIPIENT)
+        except InvitationEmailConfigError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
+        except Exception:
+            # Fail-safe: since the device/user has been created, we don't want to fail too hard
+            logger.exception("Unexpected exception while sending an email")
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
```
Comment (FirelightFlagboy) [inline parsec/backend/invite.py:463]: It's nice to prevent hard crash like that, but we don't have any log on the exception that may be raised here.

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

**Comment by TimeEngineer** [comment]
Please change the comment in rust test when you change the data

File: parsec/backend/invite.py (hunk 2 above)
Comment (vxgmichel) [reply parsec/backend/invite.py:463]: `logger.exception` logs the exception in the current context

**Review by FirelightFlagboy — APPROVED** [review]

File: parsec/backend/invite.py (hunk 2 above)
Comment (touilleMan) [reply parsec/backend/invite.py:463]: `logger.exception` takes last exception automatically, so it's equivalent to:
```python
except Exception as exc:
    logger.error("foo", exc_info=exc)
```

**Pull request merged**